packages/datatrak-web-server/src/routes/EntityDescendantsRoute.ts

```diff
@@ -37,29 +37,38 @@ export class EntityDescendantsRoute extends Route<EntityDescendantsRequest> {
     let recentEntities: string[] = [];
 
     const {
-      filter: { countryCode, projectCode, grandparentId, parentId, searchString, type },
+      filter: { countryCode, projectCode, grandparentId, parentId, type, ...restOfFilter },
+      searchString,
       fields = DEFAULT_FIELDS,
     } = query;
 
+    const entityTypes = type?.split(',') ?? [];
+
     if (isLoggedIn) {
       const currentUser = await models.user.findOne({ email: session.email });
       const { recent_entities: userRecentEntities } = currentUser.preferences;
-      recentEntities = userRecentEntities?.[countryCode]?.[type] || [];
+
+      recentEntities =
+        countryCode && type
+          ? entityTypes
+              .map(
+                entityType => userRecentEntities?.[countryCode]?.[entityType as EntityType] ?? [],
+              )
+              .flat()
+          : [];
```

Comment:
Thoughts on breaking this out into a function, it feels like it's doing a lot.


I know this version is a bit more wordy, but stylistically I have a bit of a preference for doing null handling upfront rather than relying on optional chaining. I think I just find it's clearer what required.
eg.
```
const getRecentEntities = (currentUser: User, countryCode: string | undefined, type: string | undefined) => {
  const { recent_entities: userRecentEntities } = currentUser.preferences;
  if (!userRecentEntities || !countryCode || !type) {
    return [];
  }

  const recentEntitiesForCountry = userRecentEntities[countryCode];
  if (!recentEntitiesForCountry) {
    return [];
  }

  const entityTypes = type.split(',');
  const recentEntitiesOfTypes = entityTypes.map(entityType => recentEntitiesForCountry[entityType] || []).flat();
  return recentEntitiesOfTypes; 
}
```